internal/gatewayapi/route.go

```diff
@@ -1160,3 +1189,98 @@ func getIREndpointsFromEndpointSlice(endpointSlices []*discoveryv1.EndpointSlice
 
 	return endpoints
 }
+
+func GetTargetBackendReference(backendRef gwapiv1a1.BackendRef, namespace string) gwapiv1a1.PolicyTargetReferenceWithSectionName {
+	ref := gwapiv1a1.PolicyTargetReferenceWithSectionName{
+		PolicyTargetReference: gwapiv1a1.PolicyTargetReference{
+			Group:     *backendRef.Group,
+			Kind:      *backendRef.Kind,
+			Name:      backendRef.Name,
+			Namespace: NamespacePtr(NamespaceDerefOr(backendRef.Namespace, namespace)),
+		},
+		SectionName: SectionNamePtr(strconv.Itoa(int(*backendRef.Port))),
+	}
+	return ref
+}
+
+func TargetMatched(policy gwapiv1a1.BackendTLSPolicy, target gwapiv1a1.PolicyTargetReferenceWithSectionName) bool {
+
+	policyTarget := policy.Spec.TargetRef
+
+	if target.Group == policyTarget.Group &&
+		target.Kind == policyTarget.Kind &&
+		target.Name == policyTarget.Name &&
+		NamespaceDerefOr(policyTarget.Namespace, policy.Namespace) == string(*target.Namespace) {
+		if policyTarget.SectionName != nil && *policyTarget.SectionName != *target.SectionName {
+			return false
+		}
+		return true
+	}
+	return false
+}
+
+func getBackendTLSPolicy(policies []*gwapiv1a1.BackendTLSPolicy, target gwapiv1a1.PolicyTargetReferenceWithSectionName) *gwapiv1a1.BackendTLSPolicy {
+	for _, policy := range policies {
+		if TargetMatched(*policy, target) {
+			return policy
+		}
+	}
+	return nil
+}
+
+func getBackendTLSBundle(policies []*gwapiv1a1.BackendTLSPolicy, configmaps []*corev1.ConfigMap, secrets []*corev1.Secret, targetBackend gwapiv1a1.PolicyTargetReferenceWithSectionName) (error, *ir.TLSBundle) {
+
+	backendTLSPolicy := getBackendTLSPolicy(policies, targetBackend)
+
+	if backendTLSPolicy == nil {
+		return nil, nil
+	}
+
+	tlsBundle := &ir.TLSBundle{}
+
+	caRefMap := make(map[string]string)
+
+	for _, caRef := range backendTLSPolicy.Spec.TLS.CACertRefs {
```

Comment:
Are we going to implement `WellKnownCACerts *WellKnownCACertType` field?
https://github.com/kubernetes-sigs/gateway-api/blob/54387b1e8db24e0c3eb644fa6c48a1b9ee711b39/apis/v1alpha2/backendtlspolicy_types.go#L117C2-L117C75

Reply:
found no documentation how to map this with envoy. if you have idea please share @soulxu

Reply:
It looks like only "System" CAs are currently supported as `WellKnownCACerts`. Envoy docs [mention](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/ssl#enabling-certificate-verification) using a local container file (`trusted_ca.fileneame: /etc/ssl/certs/ca-certificates.crt`) to support system CAs. The docs also point out that the file location is different in each OS. It could be possible to infer the envoy OS from the docker image that EG is using in the deployment, and configure the filename accordingly. Another options is to restrict this feature to the default envoy image that is being used.

Reply:
we use the `-dev:latest` image on `main` but we use `distroless` for pinned versions (this is what envoy provides us today), so we'll need to make sure we get the system cert location correct for at least these two

Reply:
Sorry for reply late, yes, agree with what @guydc and @arkodg said here. It depends on the docker image we used, then pointed to the right system cert localtion.